holidays/countries/trinidad_and_tobago.py

```diff
@@ -0,0 +1,163 @@
+
+
+class TrinidadAndTobago(
+    HolidayBase, ChristianHolidays, HinduCalendarHolidays, InternationalHolidays, IslamicHolidays
+):
+    """Trinidad and Tobago holidays.
+
+    References:
+        * <https://otp.tt/trinidad-and-tobago/national-holidays-and-awards/>
+        * <https://en.wikipedia.org/wiki/Public_holidays_in_Trinidad_and_Tobago>
+        * <https://en.wikipedia.org/wiki/Trinidad_and_Tobago_Carnival>
+        * <https://en.wikipedia.org/wiki/Indian_Arrival_Day#Trinidad_and_Tobago>
+        * <https://www.timeanddate.com/holidays/trinidad/eid-al-fitr>
+        * <https://www.timeanddate.com/holidays/trinidad/diwali>
+    """
+
+    country = "TT"
+    default_language = "en_TT"
+    supported_categories = (OPTIONAL, PUBLIC)
+    # %s (estimated).
+    estimated_label = tr("%s (estimated)")
+    supported_languages = ("en_TT", "en_US")
+    # Trinidad and Tobago gained independence on August 31, 1962.
+    start_year = 1963
+
+    def __init__(self, islamic_show_estimated: bool = True, *args, **kwargs):
+        """
+        Args:
+            islamic_show_estimated:
+                Whether to add "estimated" label to Islamic holidays name
+                if holiday date is estimated.
+        """
+        ChristianHolidays.__init__(self)
+        HinduCalendarHolidays.__init__(self, cls=TrinidadAndTobagoHinduHolidays)
+        InternationalHolidays.__init__(self)
+        IslamicHolidays.__init__(
+            self, cls=TrinidadAndTobagoIslamicHolidays, show_estimated=islamic_show_estimated
+        )
+        super().__init__(*args, **kwargs)
+
+    def _populate_public_holidays(self):
+        # New Year's Day.
+        self._add_new_years_day(tr("New Year's Day"))
+
+        # Good Friday.
+        self._add_good_friday(tr("Good Friday"))
+
+        # Easter Monday.
+        self._add_easter_monday(tr("Easter Monday"))
+
+        if self._year >= 1996:
+            # Spiritual Baptist Liberation Day.
+            self._add_holiday_mar_30(tr("Spiritual Baptist Liberation Day"))
+
+            # Indian Arrival Day.
+            self._add_holiday_may_30(tr("Indian Arrival Day"))
+
+        # Corpus Christi.
+        self._add_corpus_christi_day(tr("Corpus Christi"))
+
+        if self._year >= 1973:
+            # Labor Day.
+            self._add_holiday_jun_19(tr("Labour Day"))
+
+        if self._year >= 1985:
+            # African Emancipation Day.
+            self._add_holiday_aug_1(tr("African Emancipation Day"))
+
+        # Independence Day.
+        self._add_holiday_aug_31(tr("Independence Day"))
+
+        if self._year >= 1976:
+            # Republic Day.
+            self._add_holiday_sep_24(tr("Republic Day"))
+
+        # Diwali.
+        self._add_diwali(tr("Diwali"))
+
+        # Christmas Day.
+        self._add_christmas_day(tr("Christmas Day"))
+
+        # Boxing Day.
+        self._add_christmas_day_two(tr("Boxing Day"))
+
+        # Eid al-Fitr.
+        self._add_eid_al_fitr_day(tr("Eid-Ul-Fitr"))
```

Comment:
For movable holidays, we use `dts_observed.update( ... )` instead.
```suggestion
        # Eid al-Fitr.
        dts_observed.update(self._add_eid_al_fitr_day(tr("Eid-Ul-Fitr")))

        if self.observed:
            self._populate_observed(dts_observed)
```